### DISCUSSION TITLE
gphotos-sync is not downloading all photos?

### DISCUSSION THREAD

**Discussion by IslandJohn** [gilesknap/gphotos-sync#424] [General]
Hi,

I'm using gphotos-sync as a container on my NAS, but I seem to be running into some issues.

I'm now running it daily with the following arguments: "--album-invert --album-regex '.*' --archived --flush-index --omit-album-date --progress /storage", because it wasn't downloading new images. However, even with this setting, some of the latest images are missing.

Here's the latest run logs (for example):

- 04-08 04:00:03 WARNING  gphotos-sync 3.1.2 2023-04-08 04:00:03.659705 
- 04-08 04:00:04 WARNING  Downloading Photos ... 
- 04-08 04:00:04 WARNING  Downloaded 0 Items, Failed 0, Already Downloaded 0 
- 04-08 04:00:04 WARNING  Indexing Shared (titled) Albums ... 
- 04-08 04:08:52 WARNING  Listed 10 Shared (titled) Albums ... 
- 04-08 04:08:54 WARNING  Listed 20 Shared (titled) Albums ... 
- 04-08 04:08:58 WARNING  Listed 30 Shared (titled) Albums ... 
- 04-08 04:09:01 WARNING  Listed 40 Shared (titled) Albums ... 
- 04-08 04:09:04 WARNING  Listed 50 Shared (titled) Albums ... 
- 04-08 04:09:13 WARNING  Listed 60 Shared (titled) Albums ... 
- 04-08 04:09:16 WARNING  Listed 70 Shared (titled) Albums ... 
- 04-08 04:09:18 WARNING  Listed 80 Shared (titled) Albums ... 
- 04-08 04:09:26 WARNING  Listed 90 Shared (titled) Albums ... 
- 04-08 04:09:30 WARNING  Listed 100 Shared (titled) Albums ... 
- 04-08 04:09:32 WARNING  Listed 110 Shared (titled) Albums ... 
- 04-08 04:09:32 WARNING  Listed 120 Shared (titled) Albums ... 
- 04-08 04:09:34 WARNING  Listed 130 Shared (titled) Albums ... 
- 04-08 04:09:46 WARNING  Listed 140 Shared (titled) Albums ... 
- 04-08 04:09:47 WARNING  Listed 150 Shared (titled) Albums ... 
- 04-08 04:09:47 WARNING  Listed 160 Shared (titled) Albums ... 
- 04-08 04:09:47 WARNING  Listed 170 Shared (titled) Albums ... 
- 04-08 04:09:47 WARNING  Listed 180 Shared (titled) Albums ... 
- 04-08 04:09:51 WARNING  Listed 190 Shared (titled) Albums ... 
- 04-08 04:09:52 WARNING  Listed 200 Shared (titled) Albums ... 
- 04-08 04:09:52 WARNING  Listed 210 Shared (titled) Albums ... 
- 04-08 04:09:52 WARNING  Listed 220 Shared (titled) Albums ... 
- 04-08 04:09:53 WARNING  Listed 230 Shared (titled) Albums ... 
- 04-08 04:09:55 WARNING  Listed 240 Shared (titled) Albums ... 
- 04-08 04:10:05 WARNING  Listed 250 Shared (titled) Albums ... 
- 04-08 04:10:13 WARNING  Listed 260 Shared (titled) Albums ... 
- 04-08 04:10:13 WARNING  Listed 270 Shared (titled) Albums ... 
- 04-08 04:10:15 WARNING  Listed 280 Shared (titled) Albums ... 
- 04-08 04:10:17 WARNING  Listed 290 Shared (titled) Albums ... 
- 04-08 04:10:26 WARNING  Listed 300 Shared (titled) Albums ... 
- 04-08 04:10:28 WARNING  Listed 310 Shared (titled) Albums ... 
- 04-08 04:10:32 WARNING  Listed 320 Shared (titled) Albums ... 
- 04-08 04:10:32 WARNING  Listed 330 Shared (titled) Albums ... 
- 04-08 04:10:37 WARNING  Listed 340 Shared (titled) Albums ... 
- 04-08 04:10:42 WARNING  Listed 350 Shared (titled) Albums ... 
- 04-08 04:10:44 WARNING  Listed 360 Shared (titled) Albums ... 
- 04-08 04:10:50 WARNING  Listed 370 Shared (titled) Albums ... 
- 04-08 04:10:54 WARNING  Listed 380 Shared (titled) Albums ... 
- 04-08 04:11:03 WARNING  Listed 390 Shared (titled) Albums ... 
- 04-08 04:11:15 WARNING  Listed 400 Shared (titled) Albums ... 
- 04-08 04:11:19 WARNING  Listed 410 Shared (titled) Albums ... 
- 04-08 04:11:21 WARNING  Listed 420 Shared (titled) Albums ... 
- 04-08 04:11:29 WARNING  Listed 430 Shared (titled) Albums ... 
- 04-08 04:11:36 WARNING  Listed 440 Shared (titled) Albums ... 
- 04-08 04:11:47 WARNING  Listed 450 Shared (titled) Albums ... 
- 04-08 04:11:47 WARNING  Listed 460 Shared (titled) Albums ... 
- 04-08 04:11:54 WARNING  Listed 470 Shared (titled) Albums ... 
- 04-08 04:12:02 WARNING  Indexed 477 Shared (titled) Albums 
- 04-08 04:12:02 WARNING  Indexing Albums ... 
- 04-08 04:19:56 WARNING  Listed 10 Albums ... 
- 04-08 04:20:21 WARNING  Listed 20 Albums ... 
- 04-08 04:20:52 WARNING  Listed 30 Albums ... 
- 04-08 04:21:21 WARNING  Listed 40 Albums ... 
- 04-08 04:21:38 WARNING  Listed 50 Albums ... 
- 04-08 04:21:58 WARNING  Listed 60 Albums ... 
- 04-08 04:22:19 WARNING  Listed 70 Albums ... 
- 04-08 04:22:38 WARNING  Listed 80 Albums ... 
- 04-08 04:22:53 WARNING  Listed 90 Albums ... 
- 04-08 04:22:58 WARNING  Indexed 94 Albums 
- 04-08 04:22:58 WARNING  Downloading Photos ... 
- 04-08 04:23:12 WARNING  Downloaded 2 Items, Failed 0, Already Downloaded 46275 
- 04-08 04:23:12 WARNING  Creating album folder links to media ... 
- 04-08 04:23:54 WARNING  Created 86233 new album folder links 
- 04-08 04:23:54 WARNING  Done. 

In this run, it should have really downloaded 12 of the latest photos that were missing (which is why I decided to run it with --flush-index) instead of just 2 (they happen to be the latest 2).

It almost appears as if it's downloading only images that are in albums (auto-categorized by Google by faces). Is that the intent with the --album* arguments?

**Comment by gilesknap** [comment]
Unfortunately I think you have an occurance of this: https://issuetracker.google.com/issues/129250413 which I mention in #119 (but note the dates and lack of activity on the google issue).

When I looked into this years ago it did happen to media items that were not in an album. For some reason it completely stopped happening on my media library. One possible explanation for that is that I have an 'auto' folders set up to automatically create an album with all the pictures of my dog.

Maybe you could do an experiment and set up an auto folder to see if this goes away. Also I seem to recall that simply waiting a couple of days might cause the missing items to appear in media list commands.

**Reply by IslandJohn** [reply]
Thanks. I have several auto folders, so that doesn't seem to resolve the issue. It's also odd, in that some of the pictures being downloaded are ones that don't end up in an auto-folder, but then there are ones missing that did end up in one.

**Reply by IslandJohn** [reply]
Hmm, there's something odd happening (maybe an args parsing issue?).

I was running gphotos-sync with `--album-invert --album-regex ".*" --archived --flush-index --omit-album-date --progress /storage` because I wasn't getting albums created. This now appears to cause only photos in albums to be downloaded.

Switching to `--archived --flush-index --progress /storage` seems to download all photos (the original problems), but albums are created with dates subfolders.

So I tried `--album-invert --archived --flush-index --omit-album-date --progress /storage` but this doesn't seem to list albums or create album folders. There are no log entries for album listing.

Are these the intended behaviors?

**Reply by gilesknap** [reply]
That does not sound right. I have to confess that ``--album-invert`` ``--album-regex`` ``--omit-album-date`` are all contributed features and don't get any tests in the CI. So there could easily be some conflict between these arguments.

Are you saying that without any of those ...album... args you DO get all your photos? but you want the flatter file structure for albums?

**Reply by IslandJohn** [reply]
Yes, it seems without "album" args, I get all my pictures, and albums are created with dates.

With some album args (invert, omit), it doesn't actually create the album folder. So I added the regex, but if I add that, then I only seem to get pictures that are in albums.

**Reply by gilesknap** [reply]
Understood. Will investigate.

**Reply by gregeeh** [reply]
Has there been any progress on this as I'm finding recent additions on Google Photos are not being downloaded.  Also using `--omit-album-date` option.
